edweiss-firebase/functions/src/functions/action/toggleLectureQuizResult.ts

```diff
@@ -0,0 +1,29 @@
+import LectureDisplay from 'model/lectures/lectureDoc';
+import Quizzes from 'model/quizzes';
+import { onSanitizedCall } from 'utils/firebase';
+import { CollectionOf, Collections, getDocument, getDocumentAndRef } from 'utils/firestore';
+import { Predicate, tag } from 'utils/sanitizer';
+import { fail, INVALID_COURSE_ID, ok } from 'utils/status';
+
+export const toggleLectureQuizResult = onSanitizedCall(Quizzes.Functions.toggleLectureQuizResult, {
+	courseId: tag(Predicate.isNonEmptyString, INVALID_COURSE_ID),
+	lectureEventId: Predicate.isNonEmptyString,
+	lectureId: Predicate.isNonEmptyString,
+}, async (userId, args) => {
+
+	const thisUser = await getDocument(Collections.users, userId);
+	if (thisUser?.type != "professor") {
+		return fail("not_authorized");
+	}
+
+	const [doc, ref] = await getDocumentAndRef(CollectionOf<LectureDisplay.LectureEvent>("courses/" + args.courseId + "/lectures/" + args.lectureId + "/lectureEvents"), args.lectureEventId);
```

Comment:
What if the document doesn't exist? Or gets deleted while the professor is in the view? You might want to check that before using its data as it might crash the Cloud Function without providing any feedback to the user.

Reply:
I will!